lib/nats.js

```diff
@@ -1003,6 +992,57 @@ Client.prototype.processInbound = function() {
     }
 };
 
+/**
+ * Process server updates in info object
+ * @api internal
+ */
+Client.prototype.processServerUpdate = function() {
+  var client = this;
+  if (client.info.connect_urls && client.info.connect_urls.length > 0) {
+    // parse the infos
+    var tmp = {};
+    client.info.connect_urls.forEach(function(server) {
+      var u = 'nats://' + server;
+      var s = new Server(url.parse(u));
+      // implicit servers are ones added via the info connect_urls
+      s.implicit = true;
+      tmp[s.url.href] = s;
+    });
+
+    // remove implicit servers that are no longer reported
+    var toDelete = [];
+    client.servers.forEach(function(s, index) {
+      var u = s.url.href;
+      if(s.implicit && client.currentServer.url.href !== u && tmp[u] === undefined) {
+        // server was removed
+        toDelete.push(index);
+      }
+      // remove this entry from reported
+      delete tmp[u];
+    });
+
+    // perform the deletion
+    toDelete.reverse();
+    toDelete.forEach(function(index) {
+      client.servers.splice(index, 1);
+    });
+
+    // remaining servers are new
+    var newURLs = [];
+    for(var k in tmp) {
+      if(tmp.hasOwnProperty(k)) {
```

Comment:
What does that mean?
At this point, you should add to server pool all the URLs that are in `tmp`, and detect the ones that were never seen before, so that we pass those to the `serverDiscovered` callback.
I can't say if this is what this code is doing here...